services/121-service/src/payments/fsp-integration/intersolve-visa/intersolve-visa.service.ts

```diff
@@ -796,4 +704,24 @@ export class IntersolveVisaService
       return 0;
     }
   }
+
+  public async pauseCardOrThrow(tokenCode: string, blockToken: boolean) {
+    const wallet =
+      await this.intersolveVisaChildWalletScopedRepository.findOneOrFail({
+        where: { tokenCode: Equal(tokenCode) },
+      });
+    if (wallet.isTokenBlocked === blockToken) {
+      throw new Error('Token is already in this state');
```

Comment:
Consider throwing the state that is being set, so "Cannot pause token, it is already paused." and v.v.